src/Microsoft.AspNet.Mvc.Core/Infrastructure/ObjectResultExecutor.cs

```diff
@@ -321,10 +366,10 @@ public virtual Task ExecuteAsync(ActionContext context, ObjectResult result)
         /// <returns>
         /// The selected <see cref="IOutputFormatter"/> or <c>null</c> if no formatter can write the response.
         /// </returns>
-        protected virtual IOutputFormatter SelectFormatterUsingSortedAcceptHeaders(
+        protected virtual IOutputFormatter SelectFormatterUsingAcceptHeader(
             OutputFormatterWriteContext formatterContext,
             IList<IOutputFormatter> formatters,
-            IList<MediaTypeHeaderValue> sortedAcceptHeaders)
+            IList<MediaTypeSegmentWithQuality> acceptHeader)
```

Comment:
This is due to the change of the parsing code, moving from MediaTypeSegmentWithQuality to StringSegment (ignoring the quality after we already sorted it) implies creating a new list, which I don't want to do.